test/Microsoft.AspNet.Mvc.Core.Test/SessionStateTempDataProviderTest.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Http.Interfaces;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc.Core
+{
+    public class SessionStateTempDataProviderTest
+    {
+        [Fact]
+        public void Load_NullSession_ReturnsEmptyDictionary()
+        {
+            // Arrange
+            var testProvider = new SessionStateTempDataProvider();
+
+            // Act
+            var tempDataDictionary = testProvider.LoadTempData(
+                GetHttpContext(session: null, sessionEnabled: true));
+
+            // Assert
+            Assert.Empty(tempDataDictionary);
+        }
+
+        [Fact]
+        public void Load_NonNullSession_NoSessionData_ReturnsEmptyDictionary()
+        {
+            // Arrange
+            var testProvider = new SessionStateTempDataProvider();
+
+            // Act
+            var tempDataDictionary = testProvider.LoadTempData(
+                GetHttpContext(Mock.Of<ISessionCollection>()));
+
+            // Assert
+            Assert.Empty(tempDataDictionary);
+        }
+
+        [Fact]
+        public void Save_NullSession_NullDictionary_DoesNotThrow()
+        {
+            // Arrange
+            var testProvider = new SessionStateTempDataProvider();
+
+            // Act & Assert (does not throw)
+            testProvider.SaveTempData(GetHttpContext(session: null, sessionEnabled: false), null);
+        }
+
+        [Fact]
+        public void Save_NullSession_EmptyDictionary_DoesNotThrow()
+        {
+            // Arrange
+            var testProvider = new SessionStateTempDataProvider();
+
+            // Act & Assert (does not throw)
+            testProvider.SaveTempData(
+                GetHttpContext(session: null, sessionEnabled: false), new Dictionary<string, object>());
+        }
+
+        [Fact]
+        public void Save_NullSession_NonEmptyDictionary_Throws()
+        {
+            // Arrange
+            var testProvider = new SessionStateTempDataProvider();
+
+            // Act & Assert
+            Assert.Throws<InvalidOperationException>(
+                delegate {
+                    testProvider.SaveTempData(
+                        GetHttpContext(session: null, sessionEnabled: false),
+                        new Dictionary<string, object> { { "foo", "bar" } }
+                    );
+                }
```

Comment:
#JavaFormatting #NotEvenOnce 

most of time we'd format something like this like
```foo.Bar(() => 
{
  .....
});
```